response_operations_ui/views/collection_exercise.py

```diff
@@ -38,9 +38,14 @@ def view_collection_exercise(short_name, period, error=None, ci_loaded=False, ex
     ]
 
     ce_state = ce_details['collection_exercise']['state']
-    show_set_live_button = ce_state == 'READY_FOR_REVIEW'
+    show_set_live_button = ce_state in ('READY_FOR_REVIEW', 'FAILEDVALIDATION')
     locked = ce_state in ('LIVE', 'READY_FOR_LIVE', 'EXECUTION_STARTED', 'VALIDATED', 'EXECUTED')
     processing = ce_state in ('EXECUTION_STARTED', 'EXECUTED', 'VALIDATED')
+    validation_failed = ce_state == 'FAILEDVALIDATION'
+
+    validation_errors = ce_details['collection_exercise']['validationErrors']
```

Comment:
Is it possible for ce_details['collection_exercise'] to be empty?  If it is then it can throw a KeyError (..or is it TypeError... I can't remember sorry) and it needs to be handled.

Reply:
@insacuri a KeyError would be thrown before reaching this line and handled by the [error handler](https://github.com/ONSdigital/response-operations-ui/blob/81261b366e028354709d4fe9870b3b6d482f216b/response_operations_ui/error_handlers.py#L27).

Reply:
oki dokes ^^  I figured that was the case but I've been seeing a LOT of thing['something']['somethingelse'] and can't stop worrying that on the off chance these exceptions aren't being handled, and if they are, nothing is being logged when it really should be.

Reply:
Mm. Symptomatic of passing JSON about from service to service, I suppose.